backwardcompatibilityml/helpers/training.py

```diff
@@ -355,10 +356,18 @@ def get_all_error_instance_indices(h1, h2, batched_evaluation_data, batched_eval
             instance_ground_truths = batched_evaluation_target.index_select(
                 0, torch.tensor(error_instance_ids).to(device)).tolist()
 
-        return list(zip(error_instance_ids,
-                    h1_predictions,
-                    h2_predictions,
-                    instance_ground_truths))
+        class_incompatible_instance_ids = {}
+        if len(h1h2_incompatible_ids) > 0:
+            incompatible_instance_indices = torch.tensor(batch_ids).index_select(
+                0, torch.tensor(h1h2_incompatible_ids)).tolist()
+            target_incompatible_classes = batched_evaluation_target[h1h2_incompatible_ids].view(-1).tolist()
+            for class_label in set(batched_evaluation_target.view(-1).tolist()):
+                class_incompatible_instance_ids[class_label] = []
+            for incompatible_instance_index, incompatible_class in zip(incompatible_instance_indices, target_incompatible_classes):
+                class_incompatible_instance_ids[incompatible_class].append(incompatible_instance_index)
+
+        error_instance_list = list(zip(error_instance_ids, h1_predictions, h2_predictions, instance_ground_truths))
+        return (error_instance_list, class_incompatible_instance_ids)
```

Comment:
Your calculations here are correct. However, the list of all the error instances is required by the error instances table, which shows both. So it is important that this function remain as it is.

I'd move your calculation up one level, so that the incompatible points by class is calculated from the return value of this function.